Extension/src/common.ts

```diff
@@ -342,6 +342,13 @@ export function resolveVariables(input: string | undefined, additionalEnvironmen
         return "";
     }
 
+    // jsonc parser may assign a non-string object to a string.
```

Comment:
Do you know which parse call results in the malformed object?  We shouldn't ever assume that a JSON file we parse is in the format we expect.  The data has to be validated before it can be used.  I'm worried we have an unvalidated object somewhere in the code still.

Reply:
It's not a malformed object. It repros whenever a non-string is used. The parsing code just returns and any and we blindly assign it to what we expect, i.e. a string. This fix essentially checks for that and converts it to a string.

Reply:
Malformed may not be the correct terminology.  I was assuming you parsed the json into a defined interface instead of `any`.  I'm still surprised that TypeScript would let you pass an `any` for `string` without so much as a warning.  Are we doing a type assertion before calling this function?  That's what I'm trying to point out.  That we should be doing validation somewhere instead of a type assertion.

Reply:
The thing that I don't want to see throughout the code is this:
```ts
function foo(bar: string) {
  if (typeof(bar) !== "string") {
    ...
  }
}
```
If we can't rely on the variable/argument typings, then TypeScript isn't serving us well.

Reply:
Yeah, I was looking into potentially doing the validation earlier on when the parse is done, but doing it here seems just as well...and I figured we could move the check later if we find a reason to.

Also, the jsonc library has updated to not use "any" any more, however, switching to the new API has breaking changes and so switching to that would require a significantly more amount of work, but probably recommended in the future.

Reply:
I was considering adding a comment about that, but I decided not to (seems like the comment should go at the jsonc calls, but there are multiple uses, so I wasn't sure where to add the comment).

Reply:
If you check in this fix, please open a new issue to fix the place where we did a type assertion instead of a validation.

Reply:
I filed https://github.com/microsoft/vscode-cpptools/issues/9414